modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemsXlsMVCResourceCommand.java

```diff
@@ -0,0 +1,462 @@
+		Workbook workbook) {
+
+		Font font = workbook.createFont();
+
+		font.setBold(bold);
+		font.setFontHeightInPoints(heightInPoints);
+		font.setFontName(fontName);
+
+		CellStyle style = workbook.createCellStyle();
+
+		style.setFont(font);
+
+		return style;
+	}
+
+	private void _createDataRow(
+		ResourceRequest resourceRequest, ResourceResponse resourceResponse,
+		Locale locale, ContentDashboardItem<?> contentDashboardItem, Row row,
+		ThemeDisplay themeDisplay) {
+
+		int cellIndex = _createBasicDataCells(
+			0, locale, row, contentDashboardItem);
+
+		if (contentDashboardItem instanceof FileEntryContentDashboardItem) {
+			cellIndex = _createFileSpecificDataCells(
+				cellIndex, locale, resourceResponse, resourceRequest, row,
+				themeDisplay, contentDashboardItem);
+		}
+
+		if (contentDashboardItem instanceof
+				JournalArticleContentDashboardItem) {
+
+			_createJournalArticleSpecificDataCells(
+				cellIndex + _FIRST_JOURNAL_SPECIFIC_FIELD_INDEX, locale,
+				resourceResponse, resourceRequest, row, themeDisplay,
+				contentDashboardItem);
+		}
+	}
+
+	private int _createFileHeaderCells(
+		Row headerRow, int headerRowCellIndex, Locale locale) {
+
+		headerRowCellIndex = _createCell(
+			headerRowCellIndex, headerRow,
+			LanguageUtil.get(locale, "description"));
+
+		headerRowCellIndex = _createCell(
+			headerRowCellIndex, headerRow,
+			LanguageUtil.get(locale, "extension"));
+
+		headerRowCellIndex = _createCell(
+			headerRowCellIndex, headerRow,
+			LanguageUtil.get(locale, "file-name"));
+
+		headerRowCellIndex = _createCell(
+			headerRowCellIndex, headerRow, LanguageUtil.get(locale, "size"));
+
+		return headerRowCellIndex;
+	}
+
+	private int _createFileSpecificDataCells(
+		int cellIndex, Locale locale, ResourceResponse resourceResponse,
+		ResourceRequest resourceRequest, Row row, ThemeDisplay themeDisplay,
+		ContentDashboardItem<?> contentDashboardItem) {
+
+		JSONObject fileSpecificDataJSONObject =
+			contentDashboardItem.getSpecificInformationJSONObject(
+				ParamUtil.getString(resourceRequest, "backURL"),
+				_portal.getLiferayPortletResponse(resourceResponse), locale,
+				themeDisplay);
+
+		if (fileSpecificDataJSONObject == null)
+
+			return cellIndex;
+
+		String fileDescription = fileSpecificDataJSONObject.get(
+			"description"
+		).toString();
+
+		cellIndex = _createCell(cellIndex, row, fileDescription);
+
+		String extension = fileSpecificDataJSONObject.get(
+			"extension"
+		).toString();
+
+		cellIndex = _createCell(cellIndex, row, extension);
+
+		String fileName = fileSpecificDataJSONObject.get(
+			"fileName"
+		).toString();
+
+		cellIndex = _createCell(cellIndex, row, fileName);
+
+		String size = fileSpecificDataJSONObject.get(
+			"size"
+		).toString();
+
+		return _createCell(cellIndex, row, size);
+	}
+
+	private void _createHeaderRow(
+		Locale locale, Sheet sheet, Workbook workbook) {
+
+		Row headerRow = sheet.createRow((short)0);
+
+		headerRow.setRowStyle(
```

Comment:
Is it a requirement to set the style? The less code the better.

Reply:
It is not a requirement right now but in the future some style will need to be applied,.. but sure, can be removed and added later